tests/cxx/elaborated_type.cc

```diff
@@ -60,6 +60,21 @@ void global_qualified_elab(class ::GlobalClass* c);
 // they must also be forward-declared.
 Elaboration::Template<int, float>* namespace_qualified_template;
 
+// Elaborated types can be treated as forward-declarations, hence no fwd-decl is
+// needed if a declaration is introduced by elaborated type.
+// A declaration "owned" by elaborated type is the one which doesn't appear
```

Comment:
Any chance of expanding this explanation of 'owned' - is the thing being owned the declaration on line 69, or is it just the tag `OwningElaborated`? This explanation sounds to me like the former, but that would mean removing lines 69 should result in `getOwnedTagDecl()` returning null which doesn't feel right. From the code I feel like the first elaborated type declaration owns the tag if it occurs before a forward declaration for the tag.

Reply:
Yes, a declaration "owned" by elaborated type is actually an implicit declaration introduced by elaborated type (in line 67 in this case). It is different from any other forward declaration. (It is AFAIU; I haven't dug into clang code deeply.) Fwd-declaration in line 69 isn't very important for this test case and could be removed. The point here is that IWYU shouldn't suggest to place a new fwd-declaration if elaborated type presence is sufficient.

I understand that "owned" declaration concept is confusing, and should probably be considered as a temporary solution (if acceptable at all).